vms/platformvm/txs/executor/camino_tx_executor.go

```diff
@@ -639,6 +645,118 @@ func (e *CaminoStandardTxExecutor) UnlockDepositTx(tx *txs.UnlockDepositTx) erro
 	return nil
 }
 
+func (e *CaminoStandardTxExecutor) ClaimRewardTx(tx *txs.ClaimRewardTx) error {
+	caminoConfig, err := e.State.CaminoConfig()
+	if err != nil {
+		return err
+	}
+
+	if !caminoConfig.LockModeBondDeposit {
+		return errWrongLockMode
+	}
+
+	if err := locked.VerifyLockMode(tx.Ins, tx.Outs, caminoConfig.LockModeBondDeposit); err != nil {
+		return err
+	}
+
+	if err := e.Tx.SyntacticVerify(e.Backend.Ctx); err != nil {
+		return err
+	}
+
+	if err := e.FlowChecker.VerifyLock(
+		tx,
+		e.State,
+		tx.Ins,
+		tx.Outs,
+		e.Tx.Creds[:len(e.Tx.Creds)-1],
+		e.Config.TxFee,
+		e.Ctx.AVAXAssetID,
+		locked.StateDeposited,
+	); err != nil {
+		return fmt.Errorf("%w: %s", errFlowCheckFailed, err)
+	}
+
+	currentTimestamp := uint64(e.State.GetTimestamp().Unix())
+	claimRewardTxID := e.Tx.ID()
+
+	for _, depositTxID := range tx.DepositTxs {
+		// getting deposit tx
+
+		signedDepositTx, txStatus, err := e.State.GetTx(depositTxID)
+		if err != nil {
+			return fmt.Errorf("%w: %s", errDepositNotFound, err)
+		}
+		if txStatus != status.Committed {
+			return fmt.Errorf("%w: %s", errDepositNotFound, "tx is not committed")
+		}
+		depositTx, ok := signedDepositTx.Unsigned.(*txs.DepositTx)
+		if !ok {
+			return fmt.Errorf("%w: %s", errDepositNotFound, errWrongTxType)
+		}
+
+		// checking deposit signatures
+
+		depositRewardsOwner, ok := depositTx.RewardsOwner.(*secp256k1fx.OutputOwners)
+		if !ok {
+			return errNotSECPOwner
+		}
+
+		if err := e.Fx.VerifyPermissionUnordered(tx, e.Tx.Creds[len(e.Tx.Creds)-1], depositRewardsOwner); err != nil {
```

Comment:
This doesn't look like to hate to be inside loop

Reply:
It uses depositRewardsOwner, which is `depositRewardsOwner, ok := depositTx.RewardsOwner.(*secp256k1fx.OutputOwners)`, and depositTx is from `for _, depositTxID := range tx.DepositTxs {`